flake/modules/home-manager.nix

```diff
@@ -78,7 +84,7 @@ in {
   };
 
   config = mkIf cfg.enable {
-    programs.nvf.finalPackage = neovimConfigured.neovim;
+    programs.nvf.finalPackage = config.programs.nvf.settings.vim.build.finalPackage;
```

Comment:
set `cfg = config.programs.nvf` in the `let in` and set this to `cfg.settings.vim.build.finalPackage`.